### PR TITLE
convert `@babel/helpers` to typescript

### PR DESCRIPTION
**Pull request by colinaaa**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            |
| Patch: Bug Fix?          |
| Major: Breaking Change?  |
| Minor: New Feature?      |
| Tests Added + Pass?      |
| Documentation PR Link    | 
| Any Dependency Changes?  |
| License                  | MIT

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/13679"></a>

### CODE REVIEW

File: packages/babel-core/src/tools/build-external-helpers.ts [hunk 1]
```diff
@@ -124,8 +124,12 @@ function buildVar(allowlist) {
   return tree;
 }
 
-function buildHelpers(body, namespace, allowlist) {
-  const getHelperReference = name => {
+function buildHelpers(
+  body: t.Statement[],
+  namespace: t.Expression,
```
Comment (fedeci) [inline packages/babel-core/src/tools/build-external-helpers.ts:129]:
```suggestion
  namespace: t.Expression | null,
```

**Comment by fedeci** [comment]
Thanks!

**Comment by colinaaa** [comment]
I'm considering changing all the helpers in `helper.js` written in template string to individual helper files in `helpers` and maybe convert to typescript, too.

This automaticly makes imported `helpers` have concrete types like:

```ts
 type helpers = {
    jsx: {
        minVersion: string;
        ast: () => t.Program;
    };
    objectSpread2: {
        minVersion: string;
        ast: () => t.Program;
    };
    typeof: {
        minVersion: string;
        ast: () => t.Program;
    };
    wrapRegExp: {
        minVersion: string;
        ast: () => t.Program;
    };
}
```
 instead of `Record<string, { minVersion: string; ast: () => t.Program; }>`

But this seems to be a lot of works to do. Should I do it in another PR? Or should we just leave it as template strings? What do you think?

**Comment by nicolo-ribaudo** [comment]
`Record<string, { minVersion: string; ast: () => t.Program; }>` is already a very good improvement over the current status.

We can move the helpers to separate files since it also have other advantages (e.g. linting), but I'd consider it low priority.

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

**Pull request merged**